### ISSUE TITLE
Javascript node vs Javascript browser

### ISSUE THREAD

**Issue by Jj1martinez** [IIC2513-2019-1/syllabus#35]
Hola
Al tratar de usar arreglos creados en Node en el javascript del browser no me reconoce que estos existan o que alguna entidad exista. Por ejemplo, 
Tengo esto en el index de Eventos 
<% events.forEach((event) => { 
.
.
. 
})%> 

Lo cual me lo reconoce perfecto dentro de index y puedo trabajar con el. Pero al utilizar 

<script type="text/javascript"> 
events.forEach((event) => { 
)}
</script> 

events me aparece que es "undefinded". Entiendo que el primero corre en nuestra consola de Node y el otro en el del cliente. Hay alguna forma de traspasar esta información, para por ejemplo, crear un arreglo en 
<% arreglo={} %> 
y esta poder utilizarla en      <script type="text/javascript"> 
??
Gracias.

**Comment by gpvidal** [comment]
Hola @Jj1martinez ,

Ojo con esto, aún no hemos visto Javascript del lado del cliente! Pero tienes que tener claro una cosa: en estos momentos las vistas las está generando el servidor, por lo que debes (aún dentro de un tag de script - debería estar en un archivo aparte) las sigue generando el servidor. Entonces, para que te funcione lo que quieres hacer debes hacer esto:

```html
<script type="text/javascript"> 
  <% events.forEach((event) => { %>
    [...]
  <% )} %>
</script> 
```

Ahora, no hemos visto que se puedan traspasar variables. Hay formas de hacerlo, como por ejemplo definir una variable con los datos que tendría:

```html
<script type="text/javascript"> 
  var myEvents = [
    <% events.forEach((event) => { %>
      <%= `${event.name},` %>
    <% ) %>
  ];
  <!-- myEvents estaría disponible -->
</script> 
```

O algo similar. Se me hace un poquito "feo" en verdad.

Saludos,

**Comment by Jj1martinez** [comment]
Ahí logro acceder al arreglo eventos, pero por ejemplo el nombre no me lo reconoce. 

```
 var myEvents=[
          Cicletada, 
          Cicletada2,
];
```

Me aparece el error Uncaught ReferenceError: Cicletada is undefined

**Comment by gpvidal** [comment]
Ah! Pero... son strings?

Deberías imprimirlo como:

```js
var myEvents=[
          'Cicletada', 
          'Cicletada2',
];
```

Saludos,

**Comment by Jj1martinez** [comment]
Osea eventos tiene varios atributos de distinto tipo (date ints y strings), hay alguna manera de que myEvents posea todos los eventos como objetos? es decir, que  myEvents sea igual a events original  ?

**Comment by gpvidal** [comment]
Sí, pero en mi opinión es torcer mucho la funcionalidad. Si aún así quieres hacerlo, puedes ver (y jugar) con (esto)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON]

Saludos,

**Comment by Jj1martinez** [comment]
vale gracias
